pyomo/util/latex_printer.py

```diff
@@ -0,0 +1,1484 @@
+            )
+        ]
+
+    descriptorDict = {}
+    if use_short_descriptors:
+        descriptorDict['minimize'] = '\\min'
+        descriptorDict['maximize'] = '\\max'
+        descriptorDict['subject to'] = '\\text{s.t.}'
+        descriptorDict['with bounds'] = '\\text{w.b.}'
+    else:
+        descriptorDict['minimize'] = '\\text{minimize}'
+        descriptorDict['maximize'] = '\\text{maximize}'
+        descriptorDict['subject to'] = '\\text{subject to}'
+        descriptorDict['with bounds'] = '\\text{with bounds}'
+
+    # In the case where just a single expression is passed, add this to the constraint list for printing
+    constraints = constraints + expressions
+
+    # Declare a visitor/walker
+    visitor = _LatexVisitor()
+
+    variableMap = ComponentMap()
+    vrIdx = 0
+    for i in range(0, len(variableList)):
+        vr = variableList[i]
+        vrIdx += 1
+        if isinstance(vr, ScalarVar):
+            variableMap[vr] = 'x_' + str(vrIdx)
+        elif isinstance(vr, IndexedVar):
+            variableMap[vr] = 'x_' + str(vrIdx)
+            for sd in vr.index_set().data():
+                vrIdx += 1
+                variableMap[vr[sd]] = 'x_' + str(vrIdx)
+        else:
+            raise DeveloperError(
+                'Variable is not a variable.  Should not happen.  Contact developers'
+            )
+    visitor.variableMap = variableMap
+
+    parameterMap = ComponentMap()
+    pmIdx = 0
+    for i in range(0, len(parameterList)):
+        vr = parameterList[i]
+        pmIdx += 1
+        if isinstance(vr, ScalarParam):
+            parameterMap[vr] = 'p_' + str(pmIdx)
+        elif isinstance(vr, IndexedParam):
+            parameterMap[vr] = 'p_' + str(pmIdx)
+            for sd in vr.index_set().data():
+                pmIdx += 1
+                parameterMap[vr[sd]] = 'p_' + str(pmIdx)
+        else:
+            raise DeveloperError(
+                'Parameter is not a parameter.  Should not happen.  Contact developers'
+            )
+    visitor.parameterMap = parameterMap
+
+    setMap = ComponentMap()
+    for i in range(0, len(setList)):
+        st = setList[i]
+        setMap[st] = 'SET' + str(i + 1)
+    visitor.setMap = setMap
+
+    # starts building the output string
+    pstr = ''
+    if not use_equation_environment:
+        pstr += '\\begin{align} \n'
+        tbSpc = 4
+        trailingAligner = '& '
+    else:
+        pstr += '\\begin{equation} \n'
+        if not isSingle:
+            pstr += '    \\begin{aligned} \n'
+            tbSpc = 8
+        else:
+            tbSpc = 4
+        trailingAligner = ''
+
+    # Iterate over the objectives and print
+    for obj in objectives:
+        try:
+            obj_template, obj_indices = templatize_fcn(obj)
+        except:
+            if throw_templatization_error:
+                raise RuntimeError(
+                    "An objective has been constructed that cannot be templatized"
+                )
```

Comment:
It would be helpful to include the name of the offending objective in this error as well.